internal/files/writer.go

```diff
@@ -72,33 +58,32 @@ func writeObject(rootDir string, reader *db.TarReader, header *tar.Header) error
 		if err != nil {
 			return fmt.Errorf("write %v to disk: %w", path, err)
 		}
-
-	case tar.TypeDir:
-		err := removePathIfNotDirectory(path)
+		err = os.Chmod(path, os.FileMode(header.Mode))
```

Comment:
Is this `Chmod` necessary if we just opened the file with the correct mode?

Reply:
It isn't if we just opened it, but if it was already there and has the wrong mode, opening it doesn't change the mode, so we need to do this explicitly after. Kind of unfortunate cause it is another syscall. I couldn't figure out a way to get the open call to tell us if it created the file or just opened an existing one, cause we only need to do this for things we didn't create